internal/core/future.rs

```diff
@@ -121,6 +131,9 @@ impl<T> JoinHandle<T> {
     pub fn abort(self) {
         self.0.aborted.store(true, atomic::Ordering::Relaxed);
     }
+    /// Checks if the task associated with this `JoinHandle` has finished.
+    pub fn is_finished(&self) -> bool { self.0.inner.lock().expect("Failed to acquire mutex on FutureInner").fut == FutureState::Finished(None)  }
```

Comment:
I suggest something like this:
`matches!(self.0.inner.lock().unwrap().fut, FutureState::Finished{..})`
So there is no need to implement a PartialEq.